backend/src/main/java/de/bund/digitalservice/ris/norms/application/service/EliService.java

```diff
@@ -0,0 +1,47 @@
+package de.bund.digitalservice.ris.norms.application.service;
+
+import de.bund.digitalservice.ris.norms.application.exception.NormNotFoundException;
+import de.bund.digitalservice.ris.norms.application.port.input.LoadNormUseCase;
+import de.bund.digitalservice.ris.norms.domain.entity.eli.ExpressionEli;
+import de.bund.digitalservice.ris.norms.domain.entity.eli.WorkEli;
+import java.time.LocalDate;
+import org.springframework.stereotype.Service;
+
+/**
+ * Service for finding the next available eli.
+ */
+@Service
+public class EliService {
+
+  private final LoadNormUseCase loadNormUseCase;
+
+  public EliService(LoadNormUseCase loadNormUseCase) {
+    this.loadNormUseCase = loadNormUseCase;
+  }
+
+  /**
+   * Finds the next available {@link ExpressionEli} for the given date and language.
+   * Uses the lowest available the version number that is not in use.
+   *
+   * @param workEli     the work eli that should be used as the base for the expression eli
+   * @param pointInTime the date of the verkündung.
+   * @param language    the language of the expression.
+   * @return a new {@link ExpressionEli} for the given date.
+   */
+  public ExpressionEli findNextExpressionEli(
+    WorkEli workEli,
+    LocalDate pointInTime,
+    String language
+  ) {
+    var expressionEli = ExpressionEli.fromWorkEli(workEli, pointInTime, 1, language);
+
+    while (true) {
+      try {
+        loadNormUseCase.loadNorm(new LoadNormUseCase.Query(expressionEli));
```

Comment:
Why not just doing a sql query to look for all entries in the norms table with an eli_expression that start with the workEli and then in the java code filtering those out with the given pointInTime and language and then finding the latest version to finally determine the next available version? This way you would only have 1 DB call. Not sure how many would be necessary on prod mode, meaning how many versions we would normally have for 1 expression. And using an index on the eli_expression column on our table would be enough probably? Now sure how many data sets we will have but won't be that many. wdyt?

Reply:
I would almost never expect there to be an existing expression for a given pointInTime as this is only ever the case if multiple amending norms change the same target norm at the same date (or we have a transitive change to the amending norm). So I think it will not lead to any problems that we might need to do multiple db calls here in such a case